### ISSUE TITLE
Question: Is `points` the same as bonded amount?

### ISSUE THREAD

**Issue by frankli-dev** [paritytech/polkadot-staking-dashboard#1746]
<img width="1147" alt="Screenshot 2023-12-15 at 8 10 52 AM" src="https://github.com/paritytech/polkadot-staking-dashboard/assets/51565705/f57f9b8c-e6b1-4c89-8a6e-905238025d76">

<img width="551" alt="Screenshot_2023-12-15_at_8_52_59_AM" src="https://github.com/paritytech/polkadot-staking-dashboard/assets/51565705/ce624024-4043-4194-b901-d96731dffe6d">

Here I can see the `points` label. I'm not sure what that means.

Is it the same as `bonded amount`?

**Comment by rossbulat** [comment]
Points are proportional to the balance of the pool unless there is a slash to the pool, in which case the same points will convert to less balance. We use `points_to_balance` runtime api to convert points to balances on staking dashboard.

**Comment by rossbulat** [comment]
@frankli-dev is this okay to close?

**Comment by frankli-dev** [comment]
@rossbulat Thanks for your answer. Really appreciated.

One question. I setup substrate network & polkadot-staking-dashboard on my local and tried to bond 100k tokens but the `bond amount` shows `0.0028341238...` that really doesn't make sense.

I checked the codebase and `membership.points` contains `100_000` but the balance with `pointsToBalance` function returns very small value.

If this is by a slash, I am not sure what slash there is. Because I just created a new network and new pool and then bonded. So not sure what kind of slashes there can be.

Can you explain further about this, please?

**Comment by rossbulat** [comment]
Wow, that certainly is not a 1:1 ratio there! It would be hard to tell what the issue is without looking at the setup, are you able to determine whether there indeed were slashes? 

The initial ratio is defined by `POINTS_TO_BALANCE_INIT_RATIO`, which is set to 1.  And the balance to points conversion logic can be found [here](https://github.com/paritytech/polkadot-sdk/blob/64d52f225e127575b51ab7fdf677783f5cd497f4/substrate/frame/nomination-pools/src/lib.rs#L2942).

@Ank4n @gpestana do you think it is worth investigating this furhter?

**Comment by frankli-dev** [comment]
@rossbulat Thanks for sharing me the reference to `points to balance` logic.
But looks like the use of the function doesn't match on polkadot-staking-dashboard
https://github.com/paritytech/polkadot-staking-dashboard/blob/main/src/contexts/Pools/PoolMemberships/index.tsx#L105

```
const balance =
          (
            await api.call.nominationPoolsApi.pointsToBalance(
              membership.poolId,
              membership.points
            )
          )?.toString() || '0';
```

The first parameter is `poolId` but the sdk requires it as `current_balance`
```
fn balance_to_point(
		current_balance: BalanceOf<T>,
		current_points: BalanceOf<T>,
		new_funds: BalanceOf<T>,
	) -> BalanceOf<T> {
```

**Comment by frankli-dev** [comment]
@rossbulat My node setup is not public. Sorry, I am not sure if I can do it

**Comment by gpestana** [comment]
@frankli-dev the issue https://github.com/paritytech/polkadot-sdk/issues/416 could be the problem in your setup, yes. Depending on the `Balance` type that is setup in your runtime.

**Comment by frankli-dev** [comment]
> @frankli-dev the issue [paritytech/polkadot-sdk#416](https://github.com/paritytech/polkadot-sdk/issues/416) could be the problem in your setup, yes. Depending on the `Balance` type that is setup in your runtime. You can double-check in your runtime if the issue is solved by applying the fix [paritytech/polkadot-sdk#2706](https://github.com/paritytech/polkadot-sdk/pull/2706) -- which is pretty simple -- and check if that fixes the issue.

I also found that fix and it solved my issue. Hope it can be merged soon.
